packages/admin-ui/src/filters/filter-listbox.tsx

```diff
@@ -1,43 +1,48 @@
 import React, { useEffect, useRef } from 'react'
-
 import { ComboboxList } from 'ariakit/combobox'
-
-import { Box, tag } from '..'
+import { createComponent, useElement } from '@vtex/admin-ui-react'
 
 import * as style from './filter.style'
 import { usePopoverContext } from './filter-popover-context'
-
-export const FilterListbox = (props: FilterListboxProps) => {
-  const { children, id } = props
-  const optionsContainerRef = useRef<HTMLDivElement>(null)
-
-  const {
-    setIsScrollableLayout,
-    state: { combobox },
-  } = usePopoverContext()
-
-  const scrollHeight = optionsContainerRef?.current?.scrollHeight ?? 0
-  const containerHeight = optionsContainerRef?.current?.clientHeight ?? 0
-
-  useEffect(() => {
-    setIsScrollableLayout(scrollHeight > containerHeight)
-  }, [scrollHeight, containerHeight])
-
-  const ariakitcomboboxState = { ...combobox, matches: [] }
-
-  return (
-    <tag.span ref={optionsContainerRef} csx={style.scrollableContainer}>
-      <Box
-        as={ComboboxList as any}
-        state={{ ...ariakitcomboboxState, visible: true }}
-        id={id}
-        csx={style.list}
-      >
-        {children}
-      </Box>
-    </tag.span>
-  )
-}
+import { Box } from '..'
+
+export const FilterListbox = createComponent<'span', FilterListboxProps>(
+  (props) => {
+    const { children, id, ...restProps } = props
+    const optionsContainerRef = useRef<HTMLDivElement>(null)
+
+    const {
+      setIsScrollableLayout,
+      state: { combobox },
+    } = usePopoverContext()
+
+    const scrollHeight = optionsContainerRef?.current?.scrollHeight ?? 0
+    const containerHeight = optionsContainerRef?.current?.clientHeight ?? 0
+
+    useEffect(() => {
+      setIsScrollableLayout(scrollHeight > containerHeight)
```

Comment:
🤔 Is this useEffect trigged only on the mount step? I just say that, because refs do not trigger re-renders...

Reply:
No, the component is updated for other reasons and grows or shrinks. Then the heights are updated and if they changed this is triggered, for checking if the component became scrollable

It does work on real time

Reply:
Actually this pr broke it 😆  thanks for making me double check

Reply:
The issue was that I was overwriting ref with undefined from the props. It's fixed and now it works